Sources/CredentialsJWT/CredentialsJWT.swift

```diff
@@ -140,11 +227,13 @@ public class CredentialsJWT<C: Claims>: CredentialsPluginProtocol {
                             Log.error("Couldn't decode claims")
                             return onFailure(nil, nil)
                     }
-                    guard let userid = dictionary[subject] as? String else {
+                    // Ensure claims contain the expected subject claim (default `sub`)
+                    guard let subjectClaim = dictionary[subject] else {
                         Log.warning("Unable to create user profile: JWT claims do not contain '\(subject)'")
                         return onFailure(nil, nil)
                     }
-                    
+                    // Convert subject claim value to a String
+                    let userid = String("\(subjectClaim)")
```

Comment:
I don't quite follow this bit.. what type is `subjectClaim`?

Reply:
It comes out of the `[String:Any]` returned by JSONSerialization when we decode the claims JSON. This works for strings, ints and even optional types (as JSON has no way to represent optionals - they are either there or not).

If a user erroneously set the subject to be a claim that was a complex type, they would get a `UserProfile.id` value that had a format they probably weren't expecting.  This isn't ideal (it would probably be better to fail), but I think it would also be fairly easy to see what had gone wrong.